mypy/types.py

```diff
@@ -3272,13 +3271,11 @@ def accept(self, visitor: TypeVisitor[T]) -> T:
         return visitor.visit_literal_type(self)
 
     def __hash__(self) -> int:
-        if self._hash == -1:
-            self._hash = hash((self.value, self.fallback))
-        return self._hash
+        return hash(self.value)
```

Comment:
Is this part actually safe? This will make `Literal["foo"]` and `Literal[b"foo"]` have same hash (I don't remember why, but we always store the literal bytes value as a string).

Reply:
There's also `Literal[0]` vs `Literal[False]`, both 0 and False hash to zero IIRC